### DISCUSSION TITLE
Dockerfile of langfuse/langfuse:2 image

### DISCUSSION THREAD

**Discussion by pparreira** [langfuse/langfuse#3105] [Support] [answered]
The company I work for will be adopting Langfuse as a tool for monitoring LLMs. We really like the tool, congratulations on the work. However, we have a problem. The infrastructure team does not accept the image provided on Docker Hub or any other repository due to security concerns. We need to download the Langfuse repository from GitHub and build it ourselves.

When I accessed the Langfuse repository, I couldn't find the Dockerfile that was used to build the langfuse/langfuse:2 image. I noticed there are two Dockerfiles in the repository, but I am looking for the Dockerfile referenced in the docker-compose.yml file. Could you assist me with this issue?

We need this to deploy the solution in the production environment.

Thanks.

**Comment by marcklingen** [comment, marked as answer]
Hi @pparreira, you can use this dockerfile to build the main langfuse application container: https://github.com/langfuse/langfuse/blob/main/web/Dockerfile

I'd suggest to only build from the `production` branch as main contains unreleased changes. Also, I'd suggest to just pull from docker hub or github container registry if possible and then push to your internal registry (this is a very common workflow I have seen across teams).

**Comment by marcklingen** [comment]
Update: we’ve added documentation on how to build Langfuse from source: https://langfuse.com/docs/deployment/self-host#build-from-source
